hyperspy/tests/io/test_hdf5.py

```diff
@@ -229,3 +230,33 @@ def test_rgba16():
         "npy_files",
         "test_rgba16.npy"))
     nt.assert_true((s.data == data).all())
+
+
+class TestLoadingOOMReadOnly:
+
+    def setUp(self):
+        s = Signal(np.empty((5, 5, 5)))
+        s.save('tmp.hdf5')
+        self.shape = (10000, 10000, 100)
+        del s
+        f = h5py.File('tmp.hdf5', model='r+')
+        s = f['Experiments/__unnamed__']
+        del s['data']
+        s.create_dataset(
+            'data',
+            shape=self.shape,
+            dtype='float64',
+            chunks=True)
+        f.close()
+
+    @nt.raises(MemoryError)
```

Comment:
That would only buy us a limited amount of time. Why not fixing the issue for eternity by using a prediction of the average installed ram vs year based in current trends?

![graph-windows-64bit-p93](https://cloud.githubusercontent.com/assets/989719/10104148/e4d6424a-639f-11e5-952a-8b92491b1706.jpg)

(Yes, this is a joke.)